### ISSUE TITLE
Density profile help

### ISSUE THREAD

**Issue by dmargarone** [fbpic/fbpic#598]
Hi guys !!

I am trying a transverse parabolic density profile with the following function but not getting any channel along r:

#==========================================
ramp_start = 10.0e-6
z1 = ramp_start + 100.0e-6
z2 = z1 + 200.0e-6
z3 = z2 + 100.0e-6

Rp = 20.0e-6 # channel radius

def dens_func( z, r ):
    """Returns relative density at position z and r"""
    # Allocate relative density
    n = np.zeros_like(z)
    # Supress density before the ramp
    n = np.where(  np.sqrt( r**2 ) <= 60.0e-6, (1.0 + (r/Rp)**2), n )                                 # to cover both -ve and +ve side of r 
    # radial size less than 60e-6 and greater than -60e-6
    #
    n = np.where( (z>=ramp_start) & (z<z1), 1.0 * (z-ramp_start)/(z1-ramp_start), n )     # linear rising ramp
    #
    n = np.where( (z>=z1) & (z<z2), 1.0 , n )                                                                   # constant region
    #    
    n = np.where( (z>=z2) & (z<z3), 1.0 * (1.0 - ((z-z2)/(z3-z2))), n )                                # linear falling ramp
    # 
    return(n)
#==========================================

**Question:** Do I need to include both z and r together in every line ?

Thanks 
Yours,
DM

**Comment by dmargarone** [comment]
Even no channel is formed if I only use the following condition in "def dens_func(zr)"

def dens_func( z, r ):
n = np.zeros_like(z)
# Supress density before the ramp
n = np.where( (z>=ramp_start), (1.0 + (r/Rp)**2), n )
return(n)

**Comment by RemiLehe** [comment]
Hi @dmargarone,
I seems to me that your last example (with the shortened, simplified function) should work. 
Could you post your full input script, so that we can understand the issue better?

**Comment by dmargarone** [comment]
Hi Remi,
Thank you for quick reply. Find the input script in the attachment. 

DM

[lwfa_script_Flattop_channel.pdf](https://github.com/fbpic/fbpic/files/8947010/lwfa_script_Flattop_channel.pdf)

**Comment by RemiLehe** [comment]
Regarding the fact that the original (longer) `dens_func` did not work, this might be because this line overwrites the `r` dependency:
```
n = np.where( (z>=ramp_start) & (z<z1), 1.0 * (z-ramp_start)/(z1-ramp_start), n ) 
```
Fundamentally, this is more related to `numpy` than FBPIC. Feel free to have a look at the `numpy` documentation for `np.where` to help debug this.

**Comment by dmargarone** [comment]
Hi Remi,

Thanks a lot for your time. You are right ... my mistake was that during plotting "vmin = -1.0e4" and this makes a uniform color of the density throughout the figure. I kept struggling with this for a long time because I could not figure out why such a simple thing not working. But only now when I changed this value to normal (e.g. -1.0e5), I could see the channel. 

For the   numpy part, I will check the documentation as you said. 

Thank you and lots of best wishes.
DM
